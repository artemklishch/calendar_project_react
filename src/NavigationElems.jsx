import React from 'react';

const NavigationElems = ({ onArrowBtns, dateTitle, onClickTodayWeek, onCreateButton }) => {
  return (
    <nav className="nav">
      <button className="nav__button" onClick={onCreateButton}>
        <img src="https://img.icons8.com/color/48/000000/plus-math.png" className="plus" alt="Plus img" />
        Create
        </button>
      <button className="nav_day" onClick={onClickTodayWeek}>Today</button>
      <div className="nav__arow" onClick={onArrowBtns}>
        <div className="nav__arow_left"></div>
        <div className="nav__arow_right"></div>
      </div>
      <div className="nav__dateMonEar">
        <h2 className="nav__dateMonEar-today">{dateTitle}</h2>
      </div>
    </nav>
  );
};
export default NavigationElems;