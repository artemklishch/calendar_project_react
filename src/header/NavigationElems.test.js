import React from 'react';
import { shallow } from 'enzyme';
import NavigationElems from './NavigationElems';

import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
Enzyme.configure({ adapter: new Adapter() });
import toJson from 'enzyme-to-json';


describe('<NavigationElems />', () => {
  it('should call function on Create button', () => {
    const onShowForm = jest.fn();
    const wrappedComponent = shallow(<NavigationElems onShowForm={onShowForm} />);
    wrappedComponent.find('.nav__button').simulate('click');
    expect(onShowForm).toBeCalled();
  })

  it('should call function on Today button', () => {
    const onClickTodayWeek = jest.fn();
    const wrappedComponent = shallow(<NavigationElems onClickTodayWeek={onClickTodayWeek} />);
    wrappedComponent.find('.nav_day').simulate('click');
    expect(onClickTodayWeek).toBeCalled();
  })

  it('should call function on arrow buttons', () => {
    const onArrowBtns = jest.fn();
    const wrappedComponent = shallow(<NavigationElems onArrowBtns={onArrowBtns} />);
    wrappedComponent.find('.nav__arow').simulate('click');
    expect(onArrowBtns).toBeCalled();
  })

  it ('should display NavigationElems component', () => {
    const wrappedComponent = shallow(<NavigationElems />);
    expect(toJson(wrappedComponent)).toMatchSnapshot();
  })

});