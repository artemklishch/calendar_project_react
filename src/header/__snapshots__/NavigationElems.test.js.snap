// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavigationElems /> should display NavigationElems component 1`] = `
<nav
  className="nav"
>
  <button
    className="nav__button"
  >
    <img
      alt="Plus img"
      className="plus"
      src="https://img.icons8.com/color/48/000000/plus-math.png"
    />
    Create
  </button>
  <button
    className="nav_day"
  >
    Today
  </button>
  <div
    className="nav__arow"
  >
    <div
      className="nav__arow_left"
    />
    <div
      className="nav__arow_right"
    />
  </div>
  <div
    className="nav__dateMonEar"
  >
    <h2
      className="nav__dateMonEar-today"
    />
  </div>
</nav>
`;
